src/spmat/block_diagonal_matrix.py

```diff
@@ -0,0 +1,136 @@
+from typing import List, Tuple
```

Comment:
I think in python 3.10 we can directly use `list` and `tuple` now :)

Reply:
we can, but the imports are sometimes nice in case you actually want to specify what type the list is. E.g. `List[int]`

Reply:
Ah interesting! Good to know, removed